Macroist-SwiftUI/Features/AddFood/ManualEntry/IngredientEntryCard/IngredientEntryCard.swift

```diff
@@ -24,13 +46,51 @@ public struct IngredientEntryCard {
             || fat.inputState == .error
         }
         
-        var name: InputField.State = .init(placeholder: "Ingredient Name (Optional)")
-        var calories: InputField.State = .init(placeholder: "Calories")
-        var carbs: InputField.State = .init(placeholder: "Protein (Optional)")
-        var protein: InputField.State = .init(placeholder: "Carbs (Optional)")
-        var fat: InputField.State = .init(placeholder: "Fat (Optional)")
-        
-        public var ingredient: Ingredient?
+        public mutating func getIngredient() throws -> Ingredient {
+            let name: String = name.text.trimWhiteSpaceAndNewline()
+            var calories: CGFloat = .init()
+            var carbs: CGFloat = .init()
+            var protein: CGFloat = .init()
+            var fat: CGFloat = .init()
+            
+            // Calories
+            if !self.calories.text.trimWhiteSpaceAndNewline().isEmpty,
```

Comment:
still super gross to me to have all of these checks, might need to consider something else in the future